src/slskd/Search/SearchService.cs

```diff
@@ -198,13 +191,25 @@ void UpdateAndSaveChanges(Search search)
                     UpdateAndSaveChanges(search);
                 }));
 
-            var soulseekSearchTask = Client.SearchAsync(
+            Task<Soulseek.Search> soulseekSearchTask;
+            try
+            {
+                soulseekSearchTask = Client.SearchAsync(
```

Comment:
this is a subtle bug that i should have been more aware of.  as you pointed out in discord, the `SearchAsync` method is actually not async under the hood, it only returns a `Task`.  most of the public methods in Soulseek.NET are like this to disambiguate the behavior of guard clauses, which are designed to throw immediately at the call site, instead of when the resulting task is awaited.

we'll want to keep an eye out for similar cases.  nice catch!